lib/index.js

```diff
@@ -1,343 +1,381 @@
-            }
-            else {
+                const errors = [];
 
-                // We've already stored this joinTable in the collector. So let's
-                // merge the schemas of that one and this one
+                const joinTables = joinTableCollector.reduce((collector, joinTable) => {
 
-                // This way you can specify an array of extras in one model's relation and
-                // a different set of extras on a different model
+                    // Dedupe the joinTables -- multiple models can reference the same joinTable,
+                    // so we might have more than one entry for the same join table
+                    // across different models.
 
-                // Merge the joiSchemas and knexSchemas
+                    const { tableName, joiSchema } = joinTable;
 
-                const describe = joiSchema.describe().children;
+                    if (!collector[tableName]) {
 
-                const keysFromDescribe = Object.keys(describe)
-                    .reduce((schema, keyName) => {
+                        const [err, joiAsKnexSchema] = Mappings.convertFuncs.joi2Knex(joiSchema, ` in model "${tableName}"`);
 
-                        const type = describe[keyName].type;
-                        schema[keyName] = Joi[type]();
-                        return schema;
-                    }, {});
+                        if (err) {
+                            errors.push(err);
+                        }
+                        else {
+                            collector[tableName] = {
+                                joiSchema,
+                                knexSchema: joiAsKnexSchema
+                            };
+                        }
+                    }
+                    else {
 
-                collector[tableName].joiSchema = collector[tableName].joiSchema.keys(keysFromDescribe);
+                        // We've already stored this joinTable in the collector. So let's
+                        // merge the schemas of that one and this one
 
+                        // This way you can specify an array of extras in one model's relation and
+                        // a different set of extras on a different model
 
-                const joiAsKnexSchema = Mappings.convertFuncs.joi2Knex(joiSchema);
+                        // Merge the joiSchemas and knexSchemas
 
-                collector[tableName].knexSchema = Object.assign(
-                    {},
-                    collector[tableName].knexSchema,
-                    joiAsKnexSchema
-                );
-            }
+                        collector[tableName].joiSchema = internals.mergeJoiCompiled(collector[tableName].joiSchema, joiSchema);
 
-            return collector;
+                        /* eslint-disable no-unused-vars */
+                        const [ignoreErr, joiAsKnexSchema] = Mappings.convertFuncs.joi2Knex(collector[tableName].joiSchema, ` in model "${tableName}"`);
```

Comment:
Why is this ignored?  If there's a good reason then you can skip that using `const [, joiAsKnexSchema] = ...`.